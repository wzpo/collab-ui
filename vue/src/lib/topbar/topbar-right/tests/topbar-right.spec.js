import { shallowMount } from '@vue/test-utils';
import TopbarRight from '../index.vue';

describe('TopbarRight', () => {
  it('should match snapshot', () => {
    const container = shallowMount(TopbarRight, {
      propsData: {
        id: 'test'
      }
    });
    expect(container.html()).toMatchSnapshot();
  });

  it('should render one TopbarRight', () => {
    const container = shallowMount(TopbarRight);

    expect(container.classes('md-top-bar__right')).toBeTruthy();
  });

  it('should add customized class name if class prop is set', () => {
    const container = shallowMount(TopbarRight, {
      propsData: {
        class: 'testClassName'
      }
    });

    expect(container.find('.testClassName').exists()).toBeTruthy();
  });

  it('should render children', () => {
    const container = shallowMount(TopbarRight, {
      slots: {
        default: `<div class="testingforTbR" />`
      }
    });

    expect(container.findAll('.testingforTbR').length).toEqual(1);
  });
});

