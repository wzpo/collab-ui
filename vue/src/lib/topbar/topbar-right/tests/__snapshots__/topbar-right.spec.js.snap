// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopbarRight should match snapshot 1`] = `<div class="md-top-bar__right" id="test"></div>`;
